services/management/external.go

```diff
@@ -111,6 +119,18 @@ func (e *ExternalService) AddExternal(ctx context.Context, req *managementpb.Add
 			return err
 		}
 
+		if !req.SkipConnectionChecks {
+			if isPushMode(req.MetricsMode) {
+				if err = e.registry.CheckConnectionToService(ctx, tx.Querier, service, row); err != nil {
+					return err
+				}
+			} else {
+				if err = e.checkPullConnectionToService(ctx, tx.Querier, service, row); err != nil {
```

Comment:
BTW, to not duplicate code in pmm-managed and pmm-agent we can send CheckConnectionRequest to pmm-agent on pmm-server. What do you think?

Reply:
On, the first look sounds crazy, but let's try :)

Reply:
For me too :-D

Reply:
Changed